packages/jest-matchers/src/matchers.js

```diff
@@ -561,114 +573,25 @@ const matchers: MatchersObject = {
-        if (!Array.isArray(received)) {
-          return false;
-        }
-
-        if (expected.length !== received.length) {
-          return false;
-        }
-
-        return expected.every((exp, i) => compare(exp, received[i]));
-      }
-
-      if (expected instanceof Date && received instanceof Date) {
-        return expected.getTime() === received.getTime();
-      }
-
-      return Object.keys(expected).every(key => {
-        if (!received.hasOwnProperty(key)) {
-          return false;
-        }
-        const exp = expected[key];
-        const act = received[key];
-        if (typeof exp === 'object' && exp !== null && act !== null) {
-          return compare(exp, act);
-        }
-        return act === exp;
-      });
-    };
-
-    // Strip properties form received object that are not present in the expected
-    // object. We need it to print the diff without adding a lot of unrelated noise.
-    const findMatchObject = (expected: Object, received: Object)  => {
-      if (Array.isArray(received)) {
-        if (!Array.isArray(expected)) {
-          return received;
-        }
-
-        if (expected.length !== received.length) {
-          return received;
-        }
-
-        return expected.map((exp, i) => findMatchObject(exp, received[i]));
-
-      } else if (received instanceof Date) {
-        return received;
-      } else if (typeof received === 'object' && received !== null && typeof expected === 'object' && expected !== null) {
-
-        const matchedObject = {};
-
-        let match = false;
-        Object.keys(expected).forEach(key => {
-          if (received.hasOwnProperty(key)) {
-            match = true;
-
-            const exp = expected[key];
-            const act = received[key];
-
-            if (typeof exp === 'object' && exp !== null) {
-              matchedObject[key] = findMatchObject(exp, act);
-            } else {
-              matchedObject[key] = act;
-            }
-          }
-        });
-
-        if (match) {
-          return matchedObject;
-        } else {
-          return received;
-        }
-      } else {
-        return received;
-      }
-    };
-
-    const pass = compare(expectedObject, receivedObject);
+    const pass = equals(receivedObject, expectedObject, [iterableEquality, subsetEquality]);
 
     const message = pass
       ? () => matcherHint('.not.toMatchObject') +
-     `\n\nExpected value not to match object:\n` +
-     `  ${printExpected(expectedObject)}` +
-     `\nReceived:\n` +
-     `  ${printReceived(receivedObject)}`
-       : () => {
-         const diffString = diff(expectedObject, findMatchObject(expectedObject, receivedObject), {
-           expand: this.expand,
-         });
-         return matcherHint('.toMatchObject') +
+        `\n\nExpected value not to match object:\n` +
+        `  ${printExpected(expectedObject)}` +
+        `\nReceived:\n` +
+        `  ${printReceived(receivedObject)}`
+     : () => {
```

Comment:
I think one there should be one more space here (and in relevant block below it), isn't it?